packages/core/src/App/Containers/NotificationsDialog/notifications-dialog.jsx

```diff
@@ -5,147 +5,154 @@ import { CSSTransition } from 'react-transition-group';
+    <React.Fragment>
+        {notifications && notifications.length ? (
+            notifications.map((item, idx) => (
+                <div className='notifications-item' key={idx}>
+                    <h2 className='notifications-item__title'>
+                        {item.type && (
+                            <Icon
+                                icon={
+                                    item.type === 'info' || item.type === 'contract_sold'
+                                        ? 'IcAlertInfo'
+                                        : `IcAlert${toTitleCase(item.type)}`
+                                }
+                                className={classNames('notifications-item__title-icon', {
+                                    [`notifications-item__title-icon--${item.type}`]: item.type,
+                                })}
+                            />
+                        )}
+                        {item.header}
+                    </h2>
+                    <div className='notifications-item__message'>{item.message}</div>
+                    <div className='notifications-item__action'>
+                        {!isEmptyObject(item.action) && (
+                            <>
+                                {item.action.route ? (
+                                    <BinaryLink
+                                        onClick={toggleDialog}
+                                        active_class='notifications-item'
+                                        className={classNames(
+                                            'dc-btn',
+                                            'dc-btn--secondary',
+                                            'notifications-item__cta-button'
+                                        )}
+                                        to={item.action.route}
+                                    >
+                                        <Text weight='bold' size='xxs'>
+                                            {item.action.text}
+                                        </Text>
+                                    </BinaryLink>
+                                ) : (
+                                    <Button
+                                        className={classNames('dc-btn--secondary', 'notifications-item__cta-button')}
+                                        onClick={item.action.onClick}
+                                    >
+                                        <Text size='xxs'>{item.action.text}</Text>
+                                    </Button>
+                                )}
+                            </>
+                        )}
+                    </div>
+                </div>
+            ))
+        ) : (
+            <EmptyNotification />
+        )}
+    </React.Fragment>
+);
+
+const NotificationsDialogEl = React.forwardRef(({ notifications, toggleDialog }, ref) => {
+    const is_empty = !(notifications && notifications.length);
```

Comment:
```suggestion
    const is_empty = !notifications?.length;
```

Reply:
Same here, no need for optional chaining on `length` property.

Reply:
@salar-fs , @aaron-binary 
Thanks for the comments 👍 , I agree, we could check only notifications length 
and also as is_empty checked on NotificationsDialogEl component, I moved EmptyNotification insidethat  and totally remove this statement on the NotificationsList component